pkg/provider/aws/common.go

```diff
@@ -0,0 +1,165 @@
+// Copyright © 2024 Bank-Vaults Maintainers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/secretsmanager"
+	"github.com/aws/aws-sdk-go/service/ssm"
+)
+
+func getDataFromStore(ctx context.Context, storeClient client, storeType string, data map[string]string) (map[string]string, error) {
+	switch storeType {
+	case "sm":
+		return getDataFromSM(ctx, storeClient, data)
+	case "ssm":
+		return getDataFromSSM(ctx, storeClient, data)
+	default:
+		return nil, fmt.Errorf("unknown store type: %s", storeType)
+	}
+}
+
+func getDataFromSM(ctx context.Context, storeClient client, data map[string]string) (map[string]string, error) {
+	var secretsMap = make(map[string]string, len(data))
+	for key, value := range data {
+		if strings.Contains(value, "secretsmanager:") {
+			secret, err := storeClient.smClient.GetSecretValueWithContext(
+				ctx,
+				&secretsmanager.GetSecretValueInput{
+					SecretId: aws.String(value),
+				})
+			if err != nil {
+				return nil, fmt.Errorf("failed to get secret from AWS secrets manager: %w", err)
+			}
```

Comment:
nit: for readability, it would be cleaner if this was prefiltered (and block reduced)
```suggestion
		// sm not requested, forward env details
		if !strings.Contains(value, "secretsmanager:") {
			secretsMap[key] = value
			continue
		}
		
		// sm requested, fetch
		secret, err := storeClient.smClient.GetSecretValueWithContext(
			ctx,
			&secretsmanager.GetSecretValueInput{
				SecretId: aws.String(value),
			})
		if err != nil {
			return nil, fmt.Errorf("failed to get secret from AWS secrets manager: %w", err)
		}
		
		// rest of the code
```